### PR TITLE
Version upgrades

### PR DESCRIPTION
**Pull request by briantopping**
**How to categorize this PR?**
Use vSphere automation library for vSphere version call. Upgrade base image and misc dependencies.

/area compliance
/kind enhancement
/platform vsphere

**What this PR does / why we need it**:

**Release note**:

```other operator
Use vSphere automation library for vSphere version call. Upgrade base image and misc dependencies.
```

### CODE REVIEW

**Comment by tuxgoose** [comment]
The concourse build is supposed to fail? @briantopping

**Review by tuxgoose — APPROVED** [review]
lgtm

File: charts/images.yaml [hunk 1]
```diff
@@ -41,8 +41,8 @@ images:
 - name: csi-provisioner
   sourceRepository: github.com/kubernetes-csi/external-provisioner
   repository: registry.k8s.io/sig-storage/csi-provisioner
-  tag: v3.5.0
-  targetVersion: ">= 1.24"
+  tag: v3.6.0
+  targetVersion: ">= 1.22"
```
Comment (shafeeqes) [inline charts/images.yaml:45]: Why was this changed back, With https://github.com/gardener/gardener-extension-provider-vsphere/pull/472, we don't support Shoots < 1.24.

File: charts/images.yaml [hunk 2]
```diff
@@ -55,8 +55,8 @@ images:
 - name: csi-snapshotter
   sourceRepository: github.com/kubernetes-csi/external-snapshotter
   repository: registry.k8s.io/sig-storage/csi-snapshotter
-  tag: v6.2.2
-  targetVersion: ">= 1.24"
+  tag: v6.3.0
+  targetVersion: ">= 1.22"
```
Comment (shafeeqes) [inline charts/images.yaml:59]: Same here

File: charts/images.yaml [hunk 3]
```diff
@@ -69,8 +69,8 @@ images:
 - name: csi-snapshot-controller
   sourceRepository: github.com/kubernetes-csi/external-snapshotter
   repository: registry.k8s.io/sig-storage/snapshot-controller
-  tag: v6.2.2
-  targetVersion: ">= 1.24"
+  tag: v6.3.0
+  targetVersion: ">= 1.22"
```
Comment (shafeeqes) [inline charts/images.yaml:73]: Same here and all occurrences below

**Comment by briantopping** [comment]
@shafeeqes Bad merge on my part, thanks for catching that. I will repair and re-release in the morning.
